### PR TITLE
bump gatk4 modules

### PR DESCRIPTION
**Pull request by matthdsm**
Bumping GATK4 to v4.5.0.0

## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/gatk4/cnnscorevariants/main.nf [hunk 1]
```diff
@@ -3,7 +3,7 @@ process GATK4_CNNSCOREVARIANTS {
     label 'process_low'
 
     //Conda is not supported at the moment: https://github.com/broadinstitute/gatk/issues/7811
```
Comment (maxulysse) [inline modules/nf-core/gatk4/cnnscorevariants/main.nf:5]: Is conda fixed or still not then?

Comment (matthdsm) [reply modules/nf-core/gatk4/cnnscorevariants/main.nf:5]: haven't tested 🤷🏻  want to focus on getting the latest version available first

File: modules/nf-core/gatk4/cnnscorevariants/main.nf [hunk 2]
```diff
@@ -3,7 +3,7 @@ process GATK4_CNNSCOREVARIANTS {
...
-    container "nf-core/gatk:4.4.0.0" //Biocontainers is missing a package
+    container "nf-core/gatk:4.5.0.0" //Biocontainers is missing a package
```
Comment (maxulysse) [inline modules/nf-core/gatk4/cnnscorevariants/main.nf:6]: Was the new container pushed to quay.io?

File: modules/nf-core/gatk4/markduplicates/environment.yml [hunk 3]
```diff
@@ -4,6 +4,6 @@ channels:
   - bioconda
   - defaults
 dependencies:
-  - bioconda::gatk4=4.4.0.0
+  - bioconda::gatk4=4.5.0.0
```
Comment (maxulysse) [inline modules/nf-core/gatk4/markduplicates/environment.yml:7]: meed to update the mulled container

File: modules/nf-core/gatk4/cnnscorevariants/main.nf (hunk 2 above)
Comment (matthdsm) [reply modules/nf-core/gatk4/cnnscorevariants/main.nf:6]: No, that's what I was asking over on slack in #modules

Comment (maxulysse) [reply modules/nf-core/gatk4/cnnscorevariants/main.nf:6]: oh sorry, I checked my emails before Slack

File: modules/nf-core/gatk4/markduplicates/environment.yml (hunk 3 above)
Comment (matthdsm) [reply modules/nf-core/gatk4/markduplicates/environment.yml:7]: https://github.com/BioContainers/multi-package-containers/pull/2934

**Review by FriederikeHanssen — COMMENTED** [review]
I am unsure about the changes to the test.yaml. Rest looks great

**Comment by matthdsm** [comment]
> I am unsure about the changes to the test.yaml. Rest looks great

Yeah, I didn't do those. Those changes were made by @mashehu to get the test to work on the self hosted runners

**Review by FriederikeHanssen — APPROVED** [review]

**Comment by maxulysse** [comment]
what about `gatk4spark`?

**Comment by matthdsm** [comment]
Nice catch! I pushed the update for those too

**Comment by matthdsm** [comment]
I _think_ I fixed all the tests for now... here's hoping they pass the CI eventually

**Comment by maxulysse** [comment]
> I _think_ I fixed all the tests for now... here's hoping they pass the CI eventually

As everything was failing, I synced with master instead of relaunching the failing tests

File: modules/nf-core/gatk4/cnnscorevariants/main.nf (hunk 1 above)
Comment (matthdsm) [reply modules/nf-core/gatk4/cnnscorevariants/main.nf:5]: Just tested it, no dice, still issues with the python libs not being found

**Comment by matthdsm** [comment]
15 more jobs to go, but they seems to keep randomly failing 😢 
Is it allowed to skip some failing tests if 2 out of 3 are passing?

File: tests/config/nf-test.config [hunk 4]
```diff
@@ -31,6 +31,7 @@ profiles {
     docker {
         docker.enabled = true
         docker.userEmulation = false
```
Comment (maxulysse) [inline tests/config/nf-test.config:33]:
```suggestion
```
I'd say we can get rid of this line given it's being deprecated

Comment (matthdsm) [reply tests/config/nf-test.config]: done!

File: tests/config/nf-test.config [hunk 5]
```diff
@@ -31,7 +31,7 @@ profiles {
     docker {
         docker.enabled = true
         docker.fixOwnership = true
-        docker.runOptions = '-u $(id -u):$(id -g) --platform=linux/amd64'
```
Comment (maxulysse) [inline tests/config/nf-test.config:34]: sure about that?

Comment (matthdsm) [reply tests/config/nf-test.config:34]: It's a test for @mashehu 
https://nfcore.slack.com/archives/C050DNB6K5H/p1708330668655529

Comment (matthdsm) [reply tests/config/nf-test.config:34]: Removing the `-u $(id -u):$(id -g)` seems to have fixed the remaining docker issues on the self hosted runners for the most part

**Review by maxulysse — APPROVED** [review]

File: tests/config/nf-test.config [hunk 6]
```diff
@@ -33,6 +33,12 @@ profiles {
         docker.userEmulation = false
         docker.runOptions = '-u $(id -u):$(id -g) --platform=linux/amd64'
     }
+    docker_self_hosted{
+        docker.enabled = true
+        docker.userEmulation = false
```
Comment (mirpedrol) [inline tests/config/nf-test.config:38]: `docker.userEmulation` should be removed here too

Comment (mashehu) [reply tests/config/nf-test.config]: done, thanks

**Comment by mashehu** [comment]
this is the old version of the artifact-upload action, so you need to wait for all jobs to finish to download them

**Comment by matthdsm** [comment]
According to the logs, there's no artefacts 😭 
``` 
Run actions/upload-artifact@v3
Warning: No files were found with the provided path: /home/runner/pytest_workflow_*/*/.nextflow.log
/home/runner/pytest_workflow_*/*/log.out
/home/runner/pytest_workflow_*/*/log.err
/home/runner/pytest_workflow_*/*/work
!/home/runner/pytest_workflow_*/*/work/conda
!/home/runner/pytest_workflow_*/*/work/singularity
!/opt/actions-runner/_work/modules/modules/.singularity. No artifacts will be uploaded.
```

**Comment by mashehu** [comment]
hmm, maybe try a re-run?

**Comment by matthdsm** [comment]
Already did, singularity keeps failing for those 2 jobs

**Comment by nvnieuwk** [comment]
I think it cannot find `gcnvkernel` in singularity. Could be an issue in the environment in that container?

**Comment by matthdsm** [comment]
That's probably the case. I think the conversion from docker > singularity messes something up

**Comment by matthdsm** [comment]
@mashehu, latest commit didn't fix the artefact issue
```
Run actions/upload-artifact@v3
Warning: No files were found with the provided path: /home/ubuntu/pytest_workflow_*/*/.nextflow.log
/home/ubuntu/pytest_workflow_*/*/log.out
/home/ubuntu/pytest_workflow_*/*/log.err
/home/ubuntu/pytest_workflow_*/*/work
!/home/ubuntu/pytest_workflow_*/*/work/conda
!/home/ubuntu/pytest_workflow_*/*/work/singularity
!/opt/actions-runner/_work/modules/modules/.singularity. No artifacts will be uploaded.
```

**Comment by mashehu** [comment]
Looks like it worked (in the cases the failure was not because of the miniconda setup step) https://github.com/nf-core/modules/actions/runs/7970557972/job/21758438966?pr=4900#step:15:48

**Comment by matthdsm** [comment]
image

**Comment by matthdsm** [comment]
Thank you sooo much @mashehu

**Comment by mashehu** [comment]
Thank you for this herculean effort!
herculad-hercules-gif

**Pull request merged**